tx-pool/src/component/edges.rs

```diff
@@ -27,8 +31,22 @@ impl Edges {
         self.deps.len()
     }
 
-    pub(crate) fn insert_input(&mut self, out_point: OutPoint, txid: ProposalShortId) {
+    pub(crate) fn insert_input(
+        &mut self,
+        out_point: OutPoint,
+        txid: ProposalShortId,
+    ) -> Result<(), Reject> {
+        // inputs is occupied means double speanding happened here
+        if let Some(old) = self.inputs.get(&out_point) {
```

Comment:
suggest to use `hashmap#entry` which will reduce one hash op for the key.
```
match self.inputs.entry(out_point) {
  Entry::Occupied(occupied) => {
    return Err(Reject::RBFRejected(msg));
  }
  Entry::Vacant(vacant) => {
    vacant.insert(txid);
  }
}

Reply:
Good point, fixed!